plugins/generated-creature-renamer.cpp

```diff
@@ -0,0 +1,159 @@
+#include "Console.h"
+#include "Core.h"
+#include "DataDefs.h"
+#include "Export.h"
+#include "PluginManager.h"
+#include "df/world.h"
+#include "df/world_raws.h"
+#include "df/creature_raw.h"
+#include "df/caste_raw.h"
+#include "modules/World.h"
+
+//#include "df/world.h"
+
+using namespace DFHack;
+
+DFHACK_PLUGIN("generated-creature-renamer");
+REQUIRE_GLOBAL(world);
+
+command_result list_creatures(color_ostream &out, std::vector <std::string> & parameters);
+
+DFhackCExport command_result plugin_init(color_ostream &out, std::vector <PluginCommand> &commands)
+{
+    commands.push_back(PluginCommand(
+        "list-generated",
+        "Prints a list of generated creature tokens. Use \"list-generated detailed\" to show descriptions.",
+        list_creatures,
+        false, //allow non-interactive use
+        "Prints a list of generated creature tokens. Use \"list-generated detailed\" to show descriptions."
+    ));
+    return CR_OK;
+}
+
+DFhackCExport command_result plugin_shutdown(color_ostream &out)
+{
+    return CR_OK;
+}
+
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+
+DFhackCExport command_result plugin_enable(color_ostream &out, bool enable)
+{
+    is_enabled = enable;
+    return CR_OK;
+}
+
+#define NUM_DESC 227
+std::string descriptors[NUM_DESC] = { " blob", " quadruped", " humanoid", " silverfish", " mayfly", " dragonfly",
```

Comment:
I assume the order here is important - is there anywhere else (e.g. in df-structures) where these are defined? It might be better to use enum attributes, if possible.

Also, are there any here that don't start with a space? There aren't any that stand out to me, although I could have missed some.